### PR TITLE
Upgrade to using Email::Sender

### LINKED ISSUE #1639: Can't use SMTP due to deprecated parameter in SSL.pm
Whenever I try to run send reports or try to create a new account under /auth I get this in logs:

*******************************************************************
 Using the default of SSL_verify_mode of SSL_VERIFY_NONE for client
 is deprecated! Please set SSL_verify_mode to SSL_VERIFY_PEER
 together with SSL_ca_file|SSL_ca_path for verification.
 If you really don't want to verify the certificate and keep the
 connection open to Man-In-The-Middle attacks please set
 SSL_verify_mode explicitly to SSL_VERIFY_NONE in your application.
*******************************************************************
  at /var/www/platform/fixmystreet/web/../local/lib/perl5/Email/Send/SMTP.pm line 59.

Apache log:

[Fri Feb 10 15:08:36.932178 2017] [fcgid:warn] [pid 4249:tid 140227360274176] [client 127.0.0.1:48210] mod_fcgid: stderr: [error] Caught exception in FixMyStreet::App::Controller::Auth->email_sign_in "invalid SSL_version specified at /var/www/platform/fixmystreet/web/../local/lib/perl5/IO/Socket/SSL.pm line 389.", referer: http://mywebsite.org/auth?r=my

### PR DESCRIPTION
**Pull request by dracos**
Should fix #1639. Could do with some real-world testing on Ubuntu trusty or xenial, both of which have shown to exhibit the issue.

### CODE REVIEW

**Comment by pauliuszaleckas** [comment]
I should be able to test this on Monday.

**Comment by pauliuszaleckas** [comment]
I had switched git to this branch and run bin/install_perl_modules
And got this error when sending e-mail:
To use SSL please install IO::Socket::SSL with version>=2.007 at /var/www/fixmystreet/fixmystreet/bin/../local/lib/perl5/Net/SMTP.pm line 70.

**Comment by dracos** [comment]
Thanks for trying it out – sorry, I did not realise that dependency existed. I've added it to this branch now, if you'd like to try again.

**Comment by pauliuszaleckas** [comment]
Now there is no errors. However email sending in my case fails, but "bin/send-reports --debug" doesn't show failure reason. Maybe we can use Email::Sender::Failure to print reason?
Do you know how to force email sending with bin/send-reports, because now there is some timeout and mostly I see message "report 2: skipped by sender Email (might be due to previous failed attempts?)" ?

**Comment by pauliuszaleckas** [comment]
Learned some Perl and finally managed to get the error:
error at after DATA: Relaying disallowed as "Paulius Zaleckas" <fms-report-7-3f854e71@zaleckas.lt>
So problem is that FMS uses some strange e-mail, but not the one from DO_NOT_REPLY_EMAIL or even CONTACT_EMAIL.

**Comment by dracos** [comment]
[Reply to your first message before I saw second] The code does log the Email::Sender::Failure for front-end emails (e.g. confirmation emails). I'll look at getting better logging for cron email too. I'll do some testing here when I am able, hopefully very soon. Timeout - sorry no, you have to reset the db currently, #1335 #938 refers.

[Reply to your second message] Yes, FixMyStreet uses VERP addresses for its SMTP envelope so that it can potentially handle bounces and the like. They are of the form fms-[unique-string]@[email-domain].

**Comment by pauliuszaleckas** [comment]
So SSL works OK now. I changed code to use no-reply instead of VERP.

**Comment by pauliuszaleckas** [comment]
TLS seems to be working OK too.

**Comment by dracos** [comment]
Glad to hear that. I'll do a bit more testing here and add the logging I mentioned and then get this merged. If you'd like to be able to upgrade easily when the codebase changes, please do consider making your VERP/not-reply change configurable and submitting it upstream as a PR so that others can benefit from it, thanks :)

File: perllib/FixMyStreet/Email/Sender.pm [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+Email::Send::Test).
+
+=cut
+
+sub build_default_transport {
+    if ( FixMyStreet->test_mode ) {
+        Email::Sender::Util->easy_transport(Test => {});
+    } elsif ( my $smtp_host = FixMyStreet->config('SMTP_SMARTHOST') ) {
+        my $type = FixMyStreet->config('SMTP_TYPE') || '';
+        my $port = FixMyStreet->config('SMTP_PORT') || '';
```
Comment (davea) [inline perllib/FixMyStreet/Email/Sender.pm]: `$port` doesn't get used, did you mean to pass it into `easy_transport`?

**Pull request merged**